src/__tests__/components/Table.test.js

```diff
@@ -234,4 +234,98 @@ describe('Table', () => {
         expect(lastName.sortable).toBe(false);
         expect(email.sortable).toBe(true);
     });
+
+    describe('when server pagination is used', () => {
+        let updateDataSpy;
+        let pagination = {};
+
+        beforeEach(() => {
+            updateDataSpy = jest.fn();
+            props = {
+                ...props,
+                updateData: updateDataSpy,
+            };
+            wrapper = mount(<Table { ...props } />);
+            instance = wrapper.instance();
+        });
+
+        describe('nextPage()', () => {
+            it('should call updateData with correct page when currentPage < totalPages', () => {
+                pagination = {
+                    currentPage: 1,
+                    totalPages: 2,
+                }
+                wrapper.setState({ pagination });
+                instance.nextPage();
+
+                expect(updateDataSpy).toHaveBeenCalledTimes(1);
+                expect(updateDataSpy).toHaveBeenCalledWith({page: pagination.currentPage + 1});
+            });
+
+            it('should not call updateData if currentPage >= totalPages', () => {
+                pagination = {
+                    totalPages: 2,
+                    currentPage: 2,
+                }
+                wrapper.setState({ pagination });
+                instance.nextPage();
+
+                expect(updateDataSpy).not.toHaveBeenCalled();
+            });
+        });
+
+        describe('previousPage()', () => {
+            it('should call updateData with correct page when currentPage > 1', () => {
+                pagination = {
+                    currentPage: 2,
+                    totalPages: 4,
+                }
+                wrapper.setState({ pagination });
+                instance.previousPage();
+
+                expect(updateDataSpy).toHaveBeenCalledTimes(1);
+                expect(updateDataSpy).toHaveBeenCalledWith({page: pagination.currentPage - 1});
+            });
+
+            it('should not call updateData if currentPage <= 1', () => {
+                pagination = {
+                    totalPages: 1,
+                    currentPage: 1,
+                }
+                wrapper.setState({ pagination });
+                instance.previousPage();
+
+                expect(updateDataSpy).not.toHaveBeenCalled();
+            });
+        });
+    });
+
+    describe('displayedRows', () => {
+        it('should equal row length when server pagination is used', () => {
+            jest.spyOn(tableActions, 'calculateRows').mockReturnValue([]);
+            const rows = [{ firstName: 'Paul', lastName: 'Darragh', isOpen: true }];
+            const pagination = {
+                isServerPagination: true,
+            }
+            wrapper = mount(<Table { ...props } />);
+            wrapper.setState({ pagination, rows });
```

Comment:
I think in this instance, and the test below, it makes sense to set `rows` and server registration via the `props` - as that is the normal mode of use.  Under non-test usage, a component will not have it's state set externally, so we should resist the temptation to do so here.

Reply:
Makes sense - have updated